ktor-client/ktor-client-cio/jvmAndPosix/src/io/ktor/client/engine/cio/ConnectionFactory.kt

```diff
@@ -27,7 +27,7 @@ internal class ConnectionFactory(
             addressSemaphore.acquire()
 
             try {
-                aSocket(selector).tcpNoDelay().tcp().connect(address, configuration)
+                aSocket(selector).tcp().connect(address, configuration)
```

Comment:
Could you tell me why we dropped tcpNoDelay here?

Reply:
yeah, sure:
1. because `noDelay` is true by default: [source](https://github.com/ktorio/ktor/blob/6ab2a63747a5c0d58c882155b905a3601f5927cd/ktor-network/jvmAndPosix/src/io/ktor/network/sockets/SocketOptions.kt#L161-L164)
2. because `tcpNoDelay` function does nothing here as it should be called on builder with `options: SocketOptions.TCPClientSocketOptions` to do anything: [source](https://github.com/ktorio/ktor/blob/6ab2a63747a5c0d58c882155b905a3601f5927cd/ktor-network/jvmAndPosix/src/io/ktor/network/sockets/Builders.kt#L35-L65) - so to work it should be called on result of `tcp()`, and has no effect if called on just `aSocket()`

So in this PR I've additionally deprecated `tcpNoDelay` because it's not needed anymore and it's behaviour is confusing